lib/iris/fileformats/grib/_save_rules.py

```diff
@@ -1160,17 +1160,20 @@ def product_definition_section(cube, grib):
 def data_section(cube, grib):
     # Masked data?
     if ma.isMaskedArray(cube.data):
-        fill_value = cube.fill_value
-        if fill_value is None or np.isnan(cube.fill_value):
+        if not np.isnan(cube.data.fill_value):
```

Comment:
@lbdreyer Okay ... I think I see the connective tissue now. So is this the reason for setting the `fill_value` to nan in `_message_values` when a GRIB message has nan values in it's message data payload in order to control auto-selecting an appropriate `fill_value`?